faq/slurm.inc

```diff
@@ -91,15 +93,13 @@ href=\"?category=openfabrics#ib-locked-pages-more\">this FAQ entry for
 references about Slurm</a>.";
 
 /////////////////////////////////////////////////////////////////////////
+$q[] = "My job fails when using mpirun under Slurm 20.11";
 
-$q[] = "Any issues with Slurm 2.6.3?";
+$anchor[] = "slurm-20.11-mpirun";
 
-$anchor[] = "slurm-2.6.3-issue";
-
-$a[] = "Yes. The Slurm 2.6.3, 14.03 releases have a bug in their PMI-2
-support.
-
-For the slurm-2.6 branch, it is recommended to use the latest version
-(2.6.9 as of 2014/4), which is known to work properly with pmi2.
-
-For the slurm-14.03 branch, the fix will be in 14.03.1.";
+$a[] = "Due to the change in Slurm 20.11 - steps being exclusive by default
+using mpirun when build --with-slurm in sbatch script or under salloc may
+result in orted started process being limited to subset of resources. If you
+want to continue using mpirun in those places you should consider applying
+the following patch to your local openmpi build:
+https://github.com/openpmix/prrte/commit/0288ebbc15c36e1d3c32f6d12c47237053e06101";
```

Comment:
I'm afraid that patch isn't going to apply to earlier versions of OMPI. Would you mind if I push'd a commit that updates just this paragraph? I think everything else is fine.

Reply:
sounds good

Reply:
hey @rhc54 can you take a look at the current state? I removed the info about the fixing commit in favor of `export SLURM_WHOLE` prior to mpirun call for newer versions of Slurm. I think it's better/easier for end users?

Reply:
Interesting - let me ask the obvious: why don't we just put ` SLURM_WHOLE=1` into the environment prior to making the `srun` call and/or just add the `--whole` flag to `srun`? Then the user doesn't have to do anything.

Reply:
If we add `--whole` to srun then the call will fail wih older version of Slurm -  no such option. I also don't think that it's good idea to try to hide those details from end users - it's the main reason why we're seeing the issue today so exporting the variable prior the call in mpirun may be confusing if it's get inherited by users processes - where it come from? If we unset it.. the non-designed launcher->launcher->process stack gets even more complicated.

On the other hand the addition of `-c` we just did is compatible with older versions of Slurm and is just natural consequence of --ntasks-per-node=1 meaning "I want one task with all CPUs on I've got on the node".

All together I just want to say that at least in my opinion the only good solution is to use srun with appropriate PMI to start computing processes, but I understand the historical background and difficulty of change, which is sometimes unavoidable, but... takes time.